src/openscad_gui.cc

```diff
@@ -290,6 +294,20 @@ int gui(std::vector<std::string>& inputFiles, const std::filesystem::path& origi
   }
 #endif
 
+#ifdef ENABLE_GUI_TESTS
+  // Adds a singleshot timer that will be executed when the application will be started.
+  // the timer validates taht each mainwindow respects the expected UX behavior.
+  if(gui_test != "none"){
+      QTimer::singleShot(0, [&]()
+      {
+          for(auto w : app.windowManager.getWindows()){
+              runAllTest(w);
+          }
+          app.exit(0);
```

Comment:
Ideally, should return an exit code of 0 only when all tests pass, but can be deferred to later